dockstore-webservice/src/main/java/io/dockstore/webservice/filters/DenyPlatformPartnerFilter.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2025 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.filters;
+
+import io.dockstore.webservice.core.User;
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.ws.rs.ForbiddenException;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.container.ResourceInfo;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.ext.Provider;
+import java.lang.reflect.Method;
+import java.security.Principal;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Deny any request by a "platformPartner" user to an authorized endpoint (an endpoint that requires a User) that does not have a @RolesAllowed annotation.
```

Comment:
One alternative: call this a "robot" user and reserve ""platformPartner" for an "elevated" role